File: ASCOM.Utilities/ASCOM.Utilities/CheckedMessageBox.vb
Public Class CheckedMessageBox
    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Initialise the state of the suppress dialogue checkbox
        ChkDoNotShowAgain.Checked = GetBool(SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE, SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE_DEFAULT)
        Me.CenterToParent()

    End Sub

    Private Sub BtnOk_Click(sender As Object, e As EventArgs) Handles BtnOk.Click
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub

    Private Sub ChkDoNotShowAgain_CheckedChanged(sender As Object, e As EventArgs) Handles ChkDoNotShowAgain.CheckedChanged
        'The checkbox has been clicked so record the new value
        SetName(SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE, ChkDoNotShowAgain.Checked.ToString)
    End Sub
End Class

File: ASCOM.Utilities/ASCOM.Utilities/CheckedMessageBox.cs
using System;
using System.Windows.Forms;
using static ASCOM.Utilities.Global;

namespace ASCOM.Utilities
{
    public partial class CheckedMessageBox
    {
        /// <summary>
        /// Create a checked message box
        /// </summary>
        public CheckedMessageBox()
        {

            // This call is required by the designer.
            InitializeComponent();

            // Initialise the state of the suppress dialogue checkbox
            ChkDoNotShowAgain.Checked = GetBool(SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE, SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE_DEFAULT);
            CenterToParent();

        }

        private void BtnOk_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        private void ChkDoNotShowAgain_CheckedChanged(object sender, EventArgs e)
        {
            // The checkbox has been clicked so record the new value
            SetName(SUPPRESS_ALPACA_DRIVER_ADMIN_DIALOGUE, ChkDoNotShowAgain.Checked.ToString());
        }
    }
}